packages/commercetools/composables/src/useUserAddress/index.ts

```diff
@@ -0,0 +1,52 @@
+import { UseUserAddress } from '@vue-storefront/interfaces';
+import { AddressInput } from '../types/GraphQL';
+import { addCustomerAddress, assignCustomerAddress, removeCustomerAddress, updateCustomerAddress } from '@vue-storefront/commercetools-api';
+import { useUserAddressFactory, UseUserAddressFactoryParams } from '@vue-storefront/factories';
+import { user } from '../useUser';
+import { Customer } from '@vue-storefront/commercetools-api/lib/src/types/GraphQL';
+
+const updateUser = (newUserData: Customer) => {
+  user.value = {
+    ...user.value,
+    ...newUserData
+  };
+};
+
+const params: UseUserAddressFactoryParams<AddressInput> = {
+  addresses: user.value && user.value.id ? user.value.addresses : [],
+  addAddress: async (address, type) => {
+    const { data } = await addCustomerAddress(user.value, address);
+
+    updateUser(data.updateMyCustomer);
```

Comment:
those things should happen inside the factory. we shouldn't directly mutate any properties inside factory params, just to API calls that return values that are modified inside factory. This way we can make changes to that layer in the future without changing every integration. This applies to all `updateUser` - please check other comparable factories and their implementations or just ping me if you're not sure how to approach it.

Reply:
done!